internal/goutils/cloning.go

```diff
@@ -1,4 +1,4 @@
-package handy
+package goutils
```

Comment:
Seems like `Clone` is not used at all in the codebase, can you remove?

Reply:
This method lived under `providers` package and I moved it into handy. Reason: providers is an "external" package that shouldn't be clumped by other utilitise.

Only 2 usages exist:
```
// Location: connectors/providers/utils.go

catalogCopy, err := goutils.Clone[*CatalogWrapper](catalogInstance)
providerInfo, err := goutils.Clone[ProviderInfo](pInfo)
```

Copying structs could be `goutils` level. Almost a standard/general package. In Java that would be `Cloneable` but for structs. There could be a dedicated coning package for various purposes. But I don't see the requirement for that yet. That would be the best naming but an overkill ahaha.